java/datastructures/lib/src/main/java/codechallenges/linked-list/LinkedList.java

```diff
@@ -1,16 +1,74 @@
 package datastructures.linkedlist;
+
 import datastructures.linkedlist.Node;
 
 public class LinkedList {
   Node head = null;
   Node tail = null;  // not strictly required
+  int size = 0;
+
+  public void kthFromNEnd(int k) {
+    if (k > this.size) {
+      throw new IllegalArgumentException("k is greater than the length of the linked list");
+    }
+    Node curr = this.head;
+    int pos = this.size - k;
+
+    for(int i = 0; i < pos; i++) {
+      curr = curr.next;
+    }
```

Comment:
The challenge asks to return the value of the node that is `k` from the end of the list. Your method iterates to that node, but doesn't return anything.